src/main/webapp/app/overview/courses.component.ts

```diff
@@ -125,4 +129,20 @@ export class CoursesComponent implements OnInit, OnDestroy {
     openExam(): void {
         this.router.navigate(['courses', this.nextRelevantCourseForExam?.id, 'exams', this.nextRelevantExam!.id]);
     }
+
+    setSearchValue(searchValue: string): void {
+        this.searchCourseText = searchValue;
+    }
+
+    /**
+     * Sorts the courses and changes the sort direction in the end
+     */
+    onSort(isAscending: boolean): void {
+        const sortFunction = isAscending ? sortCourses : sortCoursesDescending;
+        if (this.courses) {
+            this.regularCourses = [...sortFunction(this.regularCourses)];
```

Comment:
When extending the previous onSort function you need to parse the `isSortAscending` property, too.